src/pip/_internal/utils/urls.py

```diff
@@ -40,8 +42,12 @@ def url_to_path(url: str) -> str:
         # If we have a UNC path, prepend UNC share notation.
         netloc = "\\\\" + netloc
     else:
+        # do not include traceback as the error message should be self-explaining
+        sys.tracebacklimit = 0
         raise ValueError(
-            f"non-local file URIs are not supported on this platform: {url!r}"
+            f"{url!r} points to the domain '{netloc}'. "
+            f"Non-local file URIs are not supported on this platform. "
+            f"Did you mean to use {re.sub('(^file:)(/)+','file:/',url)}?"
```

Comment:
Can you put the `re.sub` part in a separate statement? This is a but difficult to read.

Otherwise lgtm.